neqo-http3/src/hframe.rs

```diff
@@ -908,4 +935,213 @@ mod tests {
             panic!("wrong frame type");
         }
     }
+
+    fn test_reading_frame(
```

Comment:
This looks really good from a distance, but the call sites are very hard to read.  I think that you want to turn that list of 5 booleans into an options struct (or an enum for the sending options) so that the call sites can be clearer:

```rust
enum FrameReadingTestSend {
  DataOnly,
  DataWithFin,
  DataThenFin,
}

enum FrameReadingTestExpect {
  Error,
  Incomplete,
  FrameComplete,
  FrameAndStreamComplete,
}

// Then test_reading_frame(&buf[..i], false, false, false, true, false) becomes
             test_reading_frame(&buf[..i], FrameReadingTestSend::OnlyData, FrameReadingTestExpect::Incomplete); 
```